dowhy/causal_refuters/dummy_outcome_refuter.py

```diff
@@ -163,15 +196,48 @@ def refute_estimate(self):
         dummy_estimator = copy.deepcopy(self._estimate)
         dummy_estimator.value = 0
 
-        refute.add_significance_test_results(
-            self.test_significance(dummy_estimator, sample_estimates)
-        )
+        # refute.add_significance_test_results(
+        #     self.test_significance(dummy_estimator, sample_estimates)
+        # )
 
         return refute
+
+    def data_preprocess(self):
+        data_chunks = []

```

Comment:
a better way to write this function is to add a column to the dataframe. This is exactly the treatment if discrete/binary, otherwise a discrete category created from a continuous treatment. 

Then you use group_by from pandas to create groups based on this new columns value. And then return the groups created just like you do for discrete treatment below. Should be much shorter code  and readable too.

Reply:
That makes sense. I'll need to figure out how to efficiently find the complement of the data under consideration.